lib/codegen/ApplyPasses.cpp

```diff
@@ -123,6 +123,7 @@ void generate_object_file(llvm::Module &mod, llvm::raw_ostream &os) {
 
   auto features_string = features.getString();
   auto options = TargetOptions{};
+  options.GuaranteedTailCallOpt = true;
```

Comment:
Isn't there a way to extend Dwight's PR instead or reenable this optimization?

Reply:
I'm not sure I follow what you mean - this change is doing precisely that; it's extending Dwight's PR to re-enable TCO.

Reply:
The point is that `-tailcallopt` isn't just an IR-level feature; it hooks into lower-level parts of the code generator. You can see this in the options for `opt`: `--tailcallopt` is a top-level option rather than a pass that gets run by a pass manager.